Formula/shadowenv.rb

```diff
@@ -0,0 +1,40 @@
+class Shadowenv < Formula
+  desc "Reversible directory-local environment variable manipulations"
+  homepage "https://shopify.github.io/shadowenv/"
+  url "https://github.com/Shopify/shadowenv/archive/1.0.1.tar.gz"
+  sha256 "e94203fd2e6e3f3493ff3e91f385d3de115557e080ba838b64f9cabc22efe036"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--root", prefix, "--path", "."
+    man1.install "#{buildpath}/man/man1/shadowenv.1"
+    man5.install "#{buildpath}/man/man5/shadowlisp.5"
+  end
+
+  test do
+    expected_output_lines = [
+      "__shadowenv_data=\"0000000000000000:{\\\"scalars\\\":[],\\\"lists\\\":[]}\"",
```

Comment:
One last thing: this seems very specific, which is often good but can also sometimes be a bit fragile. Maybe a simpler "match" with one or two things only?

Reply:
Each of these lines comes in response to one of the input components, which each test a different bit of functionality of the tool. The one part that was a bit fragile was the ordering, which is not important. That's why I've got them sorted in the comparison. Converting to a match would significantly weaken the test. I do think it's best as is at this point.

Reply:
I figured out a way to remove additional fragility by testing not the output of the command, but its effects on the shell environment when evaluated, which is what truly matters.